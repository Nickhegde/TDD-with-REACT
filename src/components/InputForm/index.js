import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import { STRINGS } from 'consts';

import './InputForm.scss';

export default function InputForm() {
  const { register, handleSubmit, watch, errors } = useForm();
  const onSubmit = data => console.log(data);
  const [formData, setFormData] = useState('');

  const selectCSVFile = event => {
    const formDataVal = new FormData();
    formDataVal.append(
      'file',
      event.target.files[0],
      event.target.files[0].name
    );
    console.log("{ formDataVal }", formDataVal.getAll('file'))
    setFormData({ formDataVal });
  };
  console.log("formData", formData);
  return (
    <form className='form-container' onSubmit={handleSubmit(onSubmit)}>
      <input className='form-inputs' name="firstName" placeholder="First Name" ref={register({ required: true, maxlength: 20 })} />
      <div className='error-msg'>{errors.firstName && 'First name is required'}</div>
      <input className='form-inputs' name="lastName" placeholder="Last Name" ref={register({ required: true, maxlength: 20 })} />
      <div className='error-msg'>{errors.lastName && 'Last name is required'}</div>
      <select className='form-inputs' name="gender" ref={register}>
        <option value="male">Male</option>
        <option value="female">Female</option>
      </select>
      <input className='form-inputs' name="age" type="number" placeholder="Age" ref={register({ required: true, min: 10 })} />
      <input className='form-inputs' type="tel" placeholder="Mobile number" name="mobileNumber" ref={register({ required: true, pattern: { value: /^[0-9]{10}$/, message: 'Invalid phone number' } })} />
      <div className='error-msg'>{errors.mobileNumber && (errors.mobileNumber.message || 'Mobile number is required')}</div>
      <input className='form-upload-btn' type='file' accept=".csv" name="upload-csv-file" onChange={selectCSVFile} placeholder={STRINGS.CSV_UPLOAD} />
      <button className='form-submit-btn' type="submit">{STRINGS.SUBMIT}</button>
    </form>
  );
}
