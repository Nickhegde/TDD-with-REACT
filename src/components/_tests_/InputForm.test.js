import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { InputForm } from 'components';

describe('InputForm component', () => {
  const wrapper = shallow(<InputForm />);
  it('InputForm render check ', () => {
    expect(wrapper.find('form').length === 1);
    expect(wrapper.find('input').length === 4);
    expect(wrapper.find('select').length === 1);
    expect(wrapper.find('option').length === 2);
    expect(wrapper.find('div').length === 3);
    expect(wrapper.find('button').length === 1);
    // const Btn = wrapper.find('div.App-link');
    // Btn.simulate('click');
    expect(wrapper).toMatchSnapshot();
  });

  it('InputForm inputs validation check for first name', () => {
    const firstName = wrapper.find('form').childAt(0);

  });
});