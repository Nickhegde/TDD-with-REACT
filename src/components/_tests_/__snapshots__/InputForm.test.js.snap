// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm component InputForm render check  1`] = `ShallowWrapper {}`;
